### PR TITLE
Bugfix/recover onboarding device compat

### PR DESCRIPTION
**Pull request by ak-ledger**
### 📝 Description

Remove recover options for uncompatible devices (LLM/LLD)
Remove promotions after onboarding for uncompatible devices, based on wired device (LLD only)
### ❓ Context

- **Impacted projects**: `LLM, LLD` 
- **Linked resource(s)**: [LIVE-9980] 

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [ ] **No breaking changes** 

### 📸 Demo

https://github.com/LedgerHQ/ledger-live/assets/146743014/5b114f8f-03cc-402e-a3c0-48d44c59bfad

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-9980]: https://ledgerhq.atlassian.net/browse/LIVE-9980?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/ledger-live-common/src/featureFlags/helper.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Recover compatibility check function, based on feature and device
+export function checkRecoverCompatibility(feature, deviceModelId) {
```
Comment (KVNLS) [inline libs/ledger-live-common/src/featureFlags/helper.tsx:2]: I think that you can rename the function to remove the comment above with something like:
`checkRecoverAvailabality()` or `isRecoverDisplayed()`

`Compatibility` makes us think that it's only related to the device but it's also related to the feature flag.

File: libs/ledgerjs/packages/types-live/src/feature.ts [hunk 2]
```diff
@@ -299,6 +299,11 @@ export type Feature_NewsfeedPage = Feature<{
 
 export type Feature_ProtectServicesMobile = Feature<{
   deeplink: string;
+  compatibleDevices: Array<{
+    available: boolean;
+    comingSoon: boolean;
+    name: string;
+  }>;
   onboardingRestore: {
```
Comment (KVNLS) [inline libs/ledgerjs/packages/types-live/src/feature.ts:307]: Maybe we can use a type for `compatibleDevices` & `onboardingRestore` so we can reuse them, wdyt?

**Review by KVNLS — APPROVED** [review]
Works well, perfect!

File: libs/ledgerjs/packages/types-live/src/feature.ts (hunk 2 above)
Comment (ak-ledger) [reply libs/ledgerjs/packages/types-live/src/feature.ts:307]: Depends on the team sending us the infos. Is this a format they would like to keep and reuse for other purposes ?

Comment (KVNLS) [reply libs/ledgerjs/packages/types-live/src/feature.ts:307]: I only suggest adding types so I'm sure to get your point.
``` 
  type CompatibleDevice = {
    available: boolean;
    comingSoon: boolean;
    name: string;
  };
  ...
  compatibleDevices: CompatibleDevice[];
  ```

**Pull request merged**